nft_ingester/src/api/api_impl.rs

```diff
@@ -124,12 +127,32 @@ where
         let label = "check_health";
         self.metrics.inc_requests(label);
         let latency_timer = Instant::now();
+        let mut status = Status::OK;
 
-        self.pg_client.check_health().await.map_err(|_| DasApiError::InternalDbError)?;
+        let check = match self.pg_client.check_health().await {
+            Ok(_) => {
+                Check { status: Status::OK, name: "PostgresDB".to_string(), description: None }
+            },
+            Err(_) => {
+                status = Status::Unhealthy;
+                Check {
+                    status: Status::Unhealthy,
+                    name: "PostgresDB".to_string(),
+                    description: Some(DasApiError::InternalDbError.to_string()),
+                }
+            },
+        };
+
+        let response = HealthCheckResponse {
+            status,
+            app_version: self.health_check_info.app_version.clone(),
+            node_name: self.health_check_info.node_name.clone(),
+            checks: vec![check],
+        };
```

Comment:
```
        // I would do sth like this
        
        let label = "check_health";
        self.metrics.inc_requests(label);
        let latency_timer = Instant::now();
        let mut check = Check::new("PostgresDB");

        if self.pg_client.check_health().await.is_err() {
            check.status = Status::Unhealthy(DasApiError::InternalDbError.to_string());
        }

        let response = HealthCheckResponse {
            status: check.status.clone(),
            app_version: self.health_check_info.app_version.clone(),
            node_name: self.health_check_info.node_name.clone(),
            checks: vec![check],
        };

        self.metrics.set_latency(label, latency_timer.elapsed().as_millis() as f64);

        Ok(json!(response))
```

Reply:
slightly changed current implementation, but left the same idea that we can have several errors and several checks for different services.
 
Like here for example:
```
{
  "status": “Ok”,
  "app_version": "1.2.3",
  "node_name": “prod_node-1”,
  "checks": [
    {
      "name": “PostgresDB”,
      "status": “Ok”,
      "details": null
    },
    {
      "name": “Redis”,
      "status": “Ok”,
      "details": null
    },
    {
      "name": “Rpc”,
      "status": “Ok”,
      "details": null
    },
  ]
}
```